seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java

```diff
@@ -344,6 +351,11 @@ private static int getSplitOwner(TopicPartition tp, int numReaders) {
     private Map<TopicPartition, Long> listOffsets(
             Collection<TopicPartition> partitions, OffsetSpec offsetSpec)
             throws ExecutionException, InterruptedException {
+
+        if (isStreamingMode) {
+            return Collections.emptyMap();
+        }
```

Comment:
@Carl-Zhou-CN The fix is LGTM.

But I have a question about Kafka Batch mode,  there is no option to set end offset, so how do stop it in batch mode?

Reply:
In batch processing mode, the last offset in the partition will be consumed when reaching the slice.

Reply:
OK, looks we need update the doc,  I am not find related notes.

Reply:
![image](https://github.com/user-attachments/assets/022c9e46-9377-4b07-909d-608569e245a2)

There only consume the data less than Stop offset.
If this record is on the middle of `poll` result, How to commit the offset?

Reply:
Only the offset of the current consumption data is submitted